### ISSUE TITLE
R debugger: starting debugger prompts file saving

### ISSUE THREAD

**Issue by EmilHvitfeldt** [posit-dev/positron#1728]
## Positron Version:

Positron Version: 2023.10.0 (Universal) build 1453
Code - OSS Version: 1.83.0
Commit: b781e9e92c1fe27c2160a8d5f852671f24884cb4
Date: 2023-10-30T03:03:08.650Z
Electron: 25.8.4
ElectronBuildId: undefined
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 22.6.0

## Steps to reproduce the issue:

1. make debugger happen in unsaved file.

I ran

```r
debugonce(lm)

lm(mpg ~ disp, data = mtcars)
```

https://github.com/posit-dev/positron/assets/14034784/7d5649a9-15ac-4280-afd9-b9ea2f7eac88

## What did you expect to happen?

For this prompt not to happen

## Were there any error messages in the output or Developer Tools console?

Nope

**Comment by lionel-** [comment]
I think it'd make sense to set the default to "save non-untitled files" in Positron:

<img width="468" alt="Image" src="https://github.com/user-attachments/assets/e8d19227-ed8a-4805-8dde-a806c6421d9e" />

What do you think @juliasilge?

**Comment by juliasilge** [comment]
Can we try having the R extension just contribute it for `.R` files? As a way to reduce the impact for other debuggers and similar to https://github.com/PowerShell/vscode-powershell/issues/3200, adding something like this:

```json
"[r]" : {
    "debug.saveBeforeStart": "nonUntitledEditorsInActiveGroup"
}
```

to here:

https://github.com/posit-dev/positron/blob/47dda03439f0c94bf3e47f28e8da7e9bdb516be7/extensions/positron-r/package.json#L350

**Comment by lionel-** [comment]
We discussed this internally and came to these conclusions:

Changing the default for one language has usability issues because these defaults have precedence over global settings set by the user. As a user it's surprising to find that a setting does not work in some files, and it's not obvious that the reason is a language override set by an extension. For this reason we prefer to avoid such language override set by extensions, and instead change the global default in the core, if it makes sense to do so for all languages.

We've polled Python users in our Slack channel and they don't think changing the default to `nonUntitledEditorsInActiveGroup` would be disruptive to their workflow. So we're going to do that as we think this is going to improve UX for all users of Positron.

**Comment by rodrigosf672** [comment]
@lionel- If I understand correctly, the expected behavior now is for file to be automatically saved once file is run, correct?

**Comment by juliasilge** [comment]
No, they _don't_ get saved:

> Untitled scratch files should no longer prompt a save dialog when a debug session is started (e.g. via `browser()` in R). Titled files should still be saved by default and silently when the session starts.

**Comment by rodrigosf672** [comment]
Yes, titled file is being saved by default and silently when session starts. It doesn't prompt me to save (it's automatic).
